src/main/java/com/example/restapp/dao/AbstractDAO.java

```diff
@@ -4,13 +4,13 @@
 
 public abstract class AbstractDAO {
 
-	protected BeanSession session;
-
-	public void setBeanSession(BeanSession session) {
+	private final BeanSession session;
+	
+	public AbstractDAO(BeanSession session) {
```

Comment:
This is really a matter of taste. There's no right or wrong answer here. I myself prefer the constructor approach to enforce the dependency. Then you have to call `super(session)` but the compiler will enforce that so it is not possible to forget.